### DISCUSSION TITLE
Displaying the comment added when dismissing an event

### DISCUSSION THREAD

**Discussion by ar3diu** [reflexsoar/reflex-api#205] [Ideas] [answered]
So in my environment I enforced the addition of a comment when dismissing an event. But where would I see this comment once an event has been dismissed?

![image](https://user-images.githubusercontent.com/61625853/174962640-04d50940-7867-4d22-b910-76309e3fa1fc.png)

Would it be a good idea to display this comment on the event's overview page or on the bottom bar next to the `Dismissed` tag...?
An audit log may also need to be logged for each Dismiss action.

**Comment by n3tsurge** [comment, marked as answer]
In the 2022.07.00 release it will show up in the Event Drawer when you click the View Event (magnifying glass) and will support many comments so analysts can talk back and forth or put copious notes on a single event.

Brian Carroll
________________________________
From: Andrei R ***@***.***>
Sent: Wednesday, June 22, 2022 2:51:38 AM
To: reflexsoar/reflex-api ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [reflexsoar/reflex-api] Displaying the comment added when dismissing an event (Discussion #205)


So in my environment I enforced the addition of a comment when dismissing an event. But where would I see this comment once an event has been dismissed?

[image]<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F61625853%2F174962640-04d50940-7867-4d22-b910-76309e3fa1fc.png&data=05%7C01%7Cbcarroll%40zeroonesecurity.com%7Ce2a6b909a6934843ee9908da541ba886%7Ca68b09e0c76f40ccb262ae8575779491%7C0%7C0%7C637914775020231762%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=9IVsYsEnEGRrYfVLJgneIcgJQriHio5P9mFCThQldpo%3D&reserved=0>

Would it be a good idea to display this comment on the event's overview page or at the bottom bar next to the Dismissed tag...?
An audit log may also need to be logged for each Dismiss action.

**Reply by ar3diu** [reply]
Cool. Thanks for sharing the info! :pray:

**Reply by ar3diu** [reply]
Hey @n3tsurge , on the same topic: I was looking to create a Dashboard in Kibana with ReflexSOAR Event Information like title, status and dismiss reason. But I couldn't find any information about the dismiss reason in the raw event log written in Elasticsearch. Any suggestions? :thinking:

**Comment by n3tsurge** [comment]
Its the field named dismiss_reason and dismiss_comment in the reflex-events index.  It also shows up on the Dashboard in Reflex itself

Brian Carroll
________________________________

**Reply by ar3diu** [reply]
Found the `dismiss_reason` field, but it contains only ID-type values: `c16f5859-4bc5-4661-b7e0-9931c63e51f5`
![image](https://user-images.githubusercontent.com/61625853/177767310-2b3e3502-0e5f-40d3-9d5e-32b1c06b69f6.png)

**Comment by n3tsurge** [comment]
Yeah you have to correlate it back to the reflex-close-reasons index.  Which means not dashboardable on the elastic side We do this because people can rename closed reasons and add custom ones in a future build

Brian Carroll
________________________________

**Reply by ar3diu** [reply]
Ok... I did not know about that index.
I will look for a method anyway to auto-map this information as it contributes to SOC Metrics/Reports.

Thanks for the answers!

**Comment by n3tsurge** [comment]
Yeah agreed we at some point plan to have a better reporting in dashboarding functionality inside reflex itself

Brian Carroll
________________________________

**Reply by ar3diu** [reply]
The current one looks pretty good. 
Also as a suggestion, it would be useful if the dashboard items were clickable.

**Comment by n3tsurge** [comment]
Yeah that's one of the enhancements the thing about the current charting library is a it's very hard to use and do stuff like that so I want to replace the charting library and make it a little bit easier for people to create custom charts and drill downs and that kind of stuff

Brian Carroll
________________________________

**Reply by ar3diu** [reply]
Sounds really good! You guys have a lot of experience and knowledge in all SOC-related processes so I'm more than sure you've thought of a lot of cool features.

**Reply by n3tsurge** [reply]
This whole back and forth made me realize there are two bugs that need fixing that will solve your issue...we are supposed to store just the Reason name not the UUID, we also aren't producing the SOC metrics we should be like time to act, time to dismiss, time to close, etc.

**Reply by ar3diu** [reply]
Yes, those would certainly be useful.

Happy to at least contribute to identifying bugs if not fixing them! :grimacing:
